public/app/plugins/panel/canvas/InlineEditBody.tsx

```diff
@@ -54,23 +76,60 @@ export function InlineEditBody() {
       },
       supplier
     );
-  }, [instanceState, activePanel]);
+  }, [instanceState, activePanel, activeTab]);
 
   const topLevelItemsContainerStyle = {
     marginLeft: 15,
     marginTop: 10,
   };
 
+  const onTabChange = (tab: string) => {
+    setActiveTab(tab);
+  };
+
+  const typeOptions = getElementTypes(instanceState?.scene.shouldShowAdvancedTypes).options;
+  const rootLayer: FrameState | undefined = instanceState?.layer;
+
+  const onAddItem = (sel: SelectableValue<string>) => {
+    const newItem = canvasElementRegistry.getIfExists(sel.value) ?? notFoundItem;
+    const newElementOptions = newItem.getNewOptions() as CanvasElementOptions;
+    newElementOptions.type = newItem.id;
+    if (newItem.defaultSize) {
+      newElementOptions.placement = { ...newElementOptions.placement, ...newItem.defaultSize };
+    }
+    if (rootLayer) {
+      const newElement = new ElementState(newItem, newElementOptions, rootLayer);
+      newElement.updateData(rootLayer.scene.context);
+      rootLayer.elements.push(newElement);
+      rootLayer.scene.save();
+
+      rootLayer.reinitializeMoveable();
+    }
+  };
+
   return (
     <>
-      {pane.categories.map((p) => renderOptionsPaneCategoryDescriptor(p))}
       <div style={topLevelItemsContainerStyle}>{pane.items.map((item) => item.render())}</div>
+      <div style={topLevelItemsContainerStyle}>
+        <AddLayerButton onChange={onAddItem} options={typeOptions} label={'Add item'} />
+      </div>
+      <div style={topLevelItemsContainerStyle}>
+        <TabsEditor onTabChange={onTabChange} />
+        {pane.categories.map((p) => renderOptionsPaneCategoryDescriptor(p, activeTab))}
+        {instanceState && activeTab === InlineEditTabs.SelectedElement && instanceState.selected.length === 0 && (
```

Comment:
This might be a little cleaner separated out into a separate bool to clean this up a bit